pkg/controlplane/reconcilers/endpointsadapter.go

```diff
@@ -139,42 +123,37 @@ func endpointSliceFromEndpoints(endpoints *corev1.Endpoints) *discovery.Endpoint
 			})
 		}
 
-		if allAddressesIPv6(append(subset.Addresses, subset.NotReadyAddresses...)) {
+		if allAddressesIPv6(subset.Addresses) {
 			endpointSlice.AddressType = discovery.AddressTypeIPv6
 		}
 
-		endpointSlice.Endpoints = append(endpointSlice.Endpoints, getEndpointsFromAddresses(subset.Addresses, endpointSlice.AddressType, true)...)
-		endpointSlice.Endpoints = append(endpointSlice.Endpoints, getEndpointsFromAddresses(subset.NotReadyAddresses, endpointSlice.AddressType, false)...)
+		endpointSlice.Endpoints = append(endpointSlice.Endpoints, getEndpointsFromAddresses(subset.Addresses, endpointSlice.AddressType)...)
```

Comment:
This seems like we're unnecessarily losing data? Why would we publish unready endpoints with the Endpoints API but not EndpointSlice?

Reply:
We _don't_ publish unready endpoints with the Endpoints API. That's why I'm removing this code; it never does anything, because `endpoints.NotReadyAddresses` is always empty.

(We _can't_ publish unready endpoints, because the `reconcilers.EndpointReconciler` interface doesn't provide any way for the apiserver to indicate to it that it's running but not ready. And if we wanted to extend it to do this in the future, we would want to be able to distinguish terminating from non-terminating as well, but there's no way to represent that in an Endpoints, so we would have to also flip the reconciler implementations around so that the EndpointSlice was their "primary" output, and the Endpoints was generated from the EndpointSlice, rather than the other way around like it is now. So even then, we wouldn't want _this_ code; we'd need _the opposite_ code, to generate `NotReadyEndpoints` in the Endpoints based on the `Conditions` in the EndpointSlice.)